Source/Scene/ColorMapExpression.js

```diff
@@ -83,13 +85,22 @@ define([
     });
 
     function setRuntime(expression) {
-        // PERFORMANCE_IDEA: this will be in dictionary mode, can we do something faster?
-        var runtimeMap = {};
         var map = expression._map;
+
+        var runtimeMap = [];
         for (var name in map) {
             if (map.hasOwnProperty(name)) {
-                var color = map[name];
-                runtimeMap[name] = Color.fromBytes(color[0], color[1], color[2]);
+                var color = Color.fromCssColorString(map[name]);
+                //>>includeStart('debug', pragmas.debug);
+                if (color === undefined) {
+                    throw new DeveloperError('color must be defined');
+                }
+                //>>includeEnd('debug');
+
+                runtimeMap.push({
+                    color : color,
+                    pattern : new RegExp(name)
```

Comment:
I'm out of the loop for what the goal of this PR is, but you can't construct a `RegExp` from any old string and have it work. If name is always guaranteed to be a plain string (i,e, not a regex itself) then you at least have to escape the string to make sure it doesn't accidentally gets processed as a regex.

If instead, name must always be a valid regex expression to start, then that problem goes away but then you have the added problem of having to escape the regex twice in the JSON (once to escape the JSON string and second to escape the regex).  It gets pretty gnarly.

If name is sometimes a regex and sometimes a plain string, then you need a separate variable or other indicator to know which it is, otherwise the problem is intractable.